src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private int size;
+    private Object[] values;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexOutOfBoundsExceptionForAdd(index);
+        growIfArrayFull();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
+        size++;
```

Comment:
Here reuse add method in order to avoid problem when index == size fails on chackIndex
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        growIfArrayFull();
        System.arraycopy(values, index, values, index + 1, size - index);
        values[index] = value;
        size++;
```